spec/data-model/README.md

```diff
@@ -46,6 +46,15 @@ the JSON and DTD definitions are intended for interchange between systems and pr
 To that end, they relax some aspects of the data model, such as allowing
 declarations, options, and attributes to be optional rather than required properties.
 
+> [!NOTE]
+> Users relying on XML representations of messages should note that
+> XML 1.0 does not allow for the representation of all C0 control characters (U+0000-U+001F)
+> and XML 1.1 does not allow for the representation of a U+0000 NULL character.
+> These characters are allowed in MessageFormat 2 messages,
+> so systems and users relying on XML as a resource format might need to
+> apply additional restrictions on message characters
+> or provide an alternative escape mechanism for these characters.
```

Comment:
I may be missing something, but these work-arounds don't address @bhaible's concern about interoperability between implementations. They recommend a path for producing valid XML within a single implementation.

Can you give an example of restrictions or escape mechanism that you had in mind here?

Reply:
A possible restriction would be to require all messages to contain no C0 control codes.

A possible escape mechanism would be to use backslash escapes like `\u0000` in the serialised XML for control codes, which would necessitate a custom parser and serialiser.

Both of these could be applied on top of what we are presenting here. Alternatively, if we accept #682, then sticking to XML 1.1 would provide built-in support for all valid MF2 messages.

Reply:
Thanks. If I understand @bhaible's feedback, both of these mechanism will render the implementation not operable with other, different implementations. In other words, if I restrict all messages to avoid C0 codes, then I cannot accept as input a message produced by an implementation which doesn't have that restriction.

I think #682 + recommending XML 1.1 would be a more robust work-around, as you say.

Reply:
> I think https://github.com/unicode-org/message-format-wg/pull/682 + recommending XML 1.1 would be a more robust work-around

When it comes to interoperability between implementations, most likely an XML document will be exchanged between the two implementations. Since XML 1.1, for XML documents, also forbids the C0 control characters (see https://www.w3.org/TR/2006/REC-xml11-20060816/#sec-well-formed definitions [1] and [2a]), there is no advantage in recommending XML 1.1. In this respect, XML 1.1 is as bad as XML 1.0.

Reply:
But XML 1.1 does allow for character reference escapes for the control characters #x1 through #x1F, yes?

Reply:
> But XML 1.1 does allow for character reference escapes for the control characters #x1 through #x1F, yes?

Yes, it does. Quoting https://www.w3.org/TR/2006/REC-xml11-20060816/#sec-xml11 :
"XML 1.1 allows the use of character references to the control characters #x1 through #x1F, most of which are forbidden in XML 1.0. For reasons of robustness, however, these characters still cannot be used directly in documents."

So, I withdraw my statement that "XML 1.1 is as bad as XML 1.0".

But trying to validate the XML 1.1 document
```
<?xml version="1.1" encoding="UTF-8" ?>
<foo>
  <bar>C0 control &#0001;</bar>
  <bar>C0 control &#0002;</bar>
  <bar>C0 control &#0003;</bar>
  <bar>C0 control &#0004;</bar>
  <bar>C0 control &#0005;</bar>
  <bar>C0 control &#0006;</bar>
  <bar>C0 control &#0007;</bar>
  <bar>C0 control &#0008;</bar>
  <bar>C0 control &#0011;</bar>
  <bar>C0 control &#0012;</bar>
  <bar>C0 control &#0014;</bar>
  <bar>C0 control &#0015;</bar>
  <bar>C0 control &#0016;</bar>
  <bar>C0 control &#0017;</bar>
  <bar>C0 control &#0018;</bar>
  <bar>C0 control &#0019;</bar>
  <bar>C0 control &#0020;</bar>
  <bar>C0 control &#0021;</bar>
  <bar>C0 control &#0022;</bar>
  <bar>C0 control &#0023;</bar>
  <bar>C0 control &#0024;</bar>
  <bar>C0 control &#0025;</bar>
  <bar>C0 control &#0026;</bar>
  <bar>C0 control &#0027;</bar>
  <bar>C0 control &#0028;</bar>
  <bar>C0 control &#0029;</bar>
  <bar>C0 control &#0030;</bar>
  <bar>C0 control &#0031;</bar>
</foo>
```
I see that all three validators that I had used in https://github.com/unicode-org/message-format-wg/issues/669#issuecomment-1952811014 don't support XML 1.1, only XML 1.0.

So, while relying on XML 1.1 would fix the problem in theory, in practice it loses, since XML 1.1 is hardly supported by software.

Even the XML support in Microsoft .NET does not support version 1.1: https://stackoverflow.com/questions/76124030/